When the app is installed from outside Google Play it crashes often with UnsatisfiedLinkError for io.objectbox.NativeLibraryLoader

I admit this may be awkward to fix, the main reason I've added it here is this crash is the most frequent one that occurs currently for production releases of the Kiwix Android when it has _not_ been installed from Google Play. It also happens less often (but still enough to appear in the top few crashes) when the app is installed from Google Play.

If we can find a way to address this problem then even those users who find other ways to install the app will be able to use it reliably.

**Reported Impact**
This crash appears to impact both users who installed the app from Google Play and those who installed it from somewhere else. It affects those who installed the app from somewhere else more frequently than those who used Google Play. It does _not_ occur in the top 40 crashes for the Chemistry and Physics simulation or WikiMed (en) apps and it's unlikely it happens in other ones; that said they may not be downloaded as frequently outside Google Play. 

There were 6 crash clusters in the first 25 rows (which are ranked in descending order of occurrences) for this crash cluster: `java.lang.UnsatisfiedLinkError io.objectbox.internal.NativeLibraryLoader.<clinit>`

|Occurrences in last 30 days in Google Play|Occurrences in last 30 days total|Total occurrences|Impacted users|URL to crash cluster|
|          ---: |          ---: |          ---: |          ---: | :---         |
|123|1990|2090|161|[Cluster 1f31cfb3](https://play.google.com/console/u/0/developers/9116215767541857492/app/4975184706939091905/vitals/crashes/1f31cfb3/details?days=30&versionCode=7230405%2C6230304%2C5230304%2C6230201%2C5230201)|
|66|411|411|14| [Cluster 1aa80b6e](https://play.google.com/console/u/0/developers/9116215767541857492/app/4975184706939091905/vitals/crashes/1aa80b6e/details?days=30&versionCode=7230405%2C6230304%2C5230304%2C6230201%2C5230201) |
|54|384|1120|120| [Cluster 6b09a04c](https://play.google.com/console/u/0/developers/9116215767541857492/app/4975184706939091905/vitals/crashes/6b09a04c/details?days=30&versionCode=7230405%2C6230304%2C5230304%2C6230201%2C5230201) |
|3|26|119|24| [Cluster 6c6e4c6c](https://play.google.com/console/u/0/developers/9116215767541857492/app/4975184706939091905/vitals/crashes/6c6e4c6c/details?days=30&versionCode=7230405%2C6230304%2C5230304%2C6230201%2C5230201) |
|0|18|22|4| [Cluster 0cd991b5](https://play.google.com/console/u/0/developers/9116215767541857492/app/4975184706939091905/vitals/crashes/0cd991b5/details?days=30&versionCode=7230405%2C6230304%2C5230304%2C6230201%2C5230201) |
|0|17|17|2| [Cluster 9b1318f4](https://play.google.com/console/u/0/developers/9116215767541857492/app/4975184706939091905/vitals/crashes/9b1318f4/details?days=30&versionCode=7230405%2C6230304%2C5230304%2C6230201%2C5230201) |


The first column is when the `Installed from Google Play` filter is not used, the second column is when that filter has been applied. The rest of the numbers are unchanged regardless of whether that filter is applied. Tip, the URL has an extra parameter appended when we filter the installs e.g. [Cluster 9b1318f4 with &installedFrom=PLAY_STORE](https://play.google.com/console/u/0/developers/9116215767541857492/app/4975184706939091905/vitals/crashes/9b1318f4/details?days=30&versionCode=7230405%2C6230304%2C5230304%2C6230201%2C5230201&installedFrom=PLAY_STORE)

The crash occurs in various app bundles (unique app versions), on various device models, and on various Android versions. It also occurs when the app is in the foreground and in the background (occasionally).

**Steps to reproduce the behavior:**
The steps to reproduce this issue are not known currently, this issue is being reported in Android Vitals, see the first screenshot with the following filters applied:

- Last 30 days
- Production releases of the app (5 were then selected automatically)
- Removed the filter for being installed from Google Play

**Screenshots**
![Screenshot 2022-07-09 at 15 17 40](https://user-images.githubusercontent.com/785891/178109697-b5837a88-58af-454a-a207-42fa6a8981b4.png)

Top crash cluster _with Installed from Google Play filter_ [Cluster 1f31cfb3](https://play.google.com/console/u/0/developers/9116215767541857492/app/4975184706939091905/vitals/crashes/1f31cfb3/details?days=30&versionCode=7230405%2C6230304%2C5230304%2C6230201%2C5230201&installedFrom=PLAY_STORE)
![Screenshot 2022-07-09 at 15 34 59](https://user-images.githubusercontent.com/785891/178110301-155a77a9-f1a8-4d54-858b-028262459a05.png)

Second crash cluster without the `Installed from Google Play` filter:
![Screenshot 2022-07-09 at 15 59 13](https://user-images.githubusercontent.com/785891/178111161-472fd18d-d81c-4fd0-b6b6-63080807d039.png)


**Expected behavior**
The app should not crash. 

That said, perhaps the best we can do is detect an issue where the app cannot find the relevant runtime file and display a message to help the end-user recognise what the problem is, or perhaps we could help them work out how to manually install the file, etc?

**Environment**
Production releases of Kiwix Android `3.4.5` at the time this bug was raised.

**Logs**
Here's an example of the most recent stacktrace for this crash from the most frequent (top) crash cluster, [Cluster 1f31cfb3](https://play.google.com/console/u/0/developers/9116215767541857492/app/4975184706939091905/vitals/crashes/1f31cfb3/details?days=30&versionCode=7230405%2C6230304%2C5230304%2C6230201%2C5230201) :

```
java.lang.LinkageError: 
  at io.objectbox.internal.NativeLibraryLoader.<clinit> (NativeLibraryLoader.java:109)
  at io.objectbox.internal.NativeLibraryLoader.ensureLoaded (NativeLibraryLoader.java)
  at io.objectbox.BoxStore.<init> (BoxStore.java:257)
  at io.objectbox.BoxStoreBuilder.build (BoxStoreBuilder.java:498)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule.providesBoxStore (DatabaseModule.java:44)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule_ProvidesBoxStoreFactory.providesBoxStore (DatabaseModule_ProvidesBoxStoreFactory.java:40)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule_ProvidesBoxStoreFactory.get (DatabaseModule_ProvidesBoxStoreFactory.java:31)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule_ProvidesBoxStoreFactory.get (DatabaseModule_ProvidesBoxStoreFactory.java:10)
  at dagger.internal.DoubleCheck.get (DoubleCheck.java:47)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule_ProvidesFetchDownloadDaoFactory.get (DatabaseModule_ProvidesFetchDownloadDaoFactory.java:35)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule_ProvidesFetchDownloadDaoFactory.get (DatabaseModule_ProvidesFetchDownloadDaoFactory.java:11)
  at dagger.internal.DoubleCheck.get (DoubleCheck.java:47)
  at org.kiwix.kiwixmobile.core.downloader.fetch.FetchDownloadMonitor_Factory.get (FetchDownloadMonitor_Factory.java:30)
  at org.kiwix.kiwixmobile.core.downloader.fetch.FetchDownloadMonitor_Factory.get (FetchDownloadMonitor_Factory.java:9)
  at org.kiwix.kiwixmobile.core.di.modules.ApplicationModule_ProvideDownloadMonitor$core_releaseFactory.get (ApplicationModule_ProvideDownloadMonitor.java:31)
  at org.kiwix.kiwixmobile.core.di.modules.ApplicationModule_ProvideDownloadMonitor$core_releaseFactory.get (ApplicationModule_ProvideDownloadMonitor.java:10)
  at dagger.internal.DoubleCheck.get (DoubleCheck.java:47)
  at org.kiwix.kiwixmobile.core.di.components.DaggerCoreComponent.injectCoreApp (DaggerCoreComponent.java:437)
  at org.kiwix.kiwixmobile.core.di.components.DaggerCoreComponent.inject (DaggerCoreComponent.java:409)
  at org.kiwix.kiwixmobile.core.CoreApp.onCreate (CoreApp.kt:85)
  at android.app.Instrumentation.callApplicationOnCreate (Instrumentation.java:1189)
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:6603)
  at android.app.ActivityThread.access$1500 (ActivityThread.java:235)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1916)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:225)
  at android.app.ActivityThread.main (ActivityThread.java:7563)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:539)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:994)
Caused by: java.lang.UnsatisfiedLinkError: 
  at java.lang.Runtime.loadLibrary0 (Runtime.java:1067)
  at java.lang.Runtime.loadLibrary0 (Runtime.java:1007)
  at java.lang.System.loadLibrary (System.java:1667)
  at io.objectbox.internal.NativeLibraryLoader.<clinit> (NativeLibraryLoader.java:84)
  at io.objectbox.internal.NativeLibraryLoader.ensureLoaded (NativeLibraryLoader.java)
  at io.objectbox.BoxStore.<init> (BoxStore.java:257)
  at io.objectbox.BoxStoreBuilder.build (BoxStoreBuilder.java:498)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule.providesBoxStore (DatabaseModule.java:44)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule_ProvidesBoxStoreFactory.providesBoxStore (DatabaseModule_ProvidesBoxStoreFactory.java:40)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule_ProvidesBoxStoreFactory.get (DatabaseModule_ProvidesBoxStoreFactory.java:31)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule_ProvidesBoxStoreFactory.get (DatabaseModule_ProvidesBoxStoreFactory.java:10)
  at dagger.internal.DoubleCheck.get (DoubleCheck.java:47)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule_ProvidesFetchDownloadDaoFactory.get (DatabaseModule_ProvidesFetchDownloadDaoFactory.java:35)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule_ProvidesFetchDownloadDaoFactory.get (DatabaseModule_ProvidesFetchDownloadDaoFactory.java:11)
  at dagger.internal.DoubleCheck.get (DoubleCheck.java:47)
  at org.kiwix.kiwixmobile.core.downloader.fetch.FetchDownloadMonitor_Factory.get (FetchDownloadMonitor_Factory.java:30)
  at org.kiwix.kiwixmobile.core.downloader.fetch.FetchDownloadMonitor_Factory.get (FetchDownloadMonitor_Factory.java:9)
  at org.kiwix.kiwixmobile.core.di.modules.ApplicationModule_ProvideDownloadMonitor$core_releaseFactory.get (ApplicationModule_ProvideDownloadMonitor.java:31)
  at org.kiwix.kiwixmobile.core.di.modules.ApplicationModule_ProvideDownloadMonitor$core_releaseFactory.get (ApplicationModule_ProvideDownloadMonitor.java:10)
  at dagger.internal.DoubleCheck.get (DoubleCheck.java:47)
  at org.kiwix.kiwixmobile.core.di.components.DaggerCoreComponent.injectCoreApp (DaggerCoreComponent.java:437)
  at org.kiwix.kiwixmobile.core.di.components.DaggerCoreComponent.inject (DaggerCoreComponent.java:409)
  at org.kiwix.kiwixmobile.core.CoreApp.onCreate (CoreApp.kt:85)
  at android.app.Instrumentation.callApplicationOnCreate (Instrumentation.java:1189)
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:6603)
  at android.app.ActivityThread.access$1500 (ActivityThread.java:235)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1916)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:225)
  at android.app.ActivityThread.main (ActivityThread.java:7563)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:539)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:994)
```

For the second most frequent crash cluster, [Cluster 1aa80b6e](https://play.google.com/console/u/0/developers/9116215767541857492/app/4975184706939091905/vitals/crashes/1aa80b6e/details?days=30&versionCode=7230405%2C6230304%2C5230304%2C6230201%2C5230201) :
```
java.lang.LinkageError: 
  at io.objectbox.internal.NativeLibraryLoader.<clinit> (NativeLibraryLoader.java:109)
  at io.objectbox.BoxStore.<init> (BoxStore.java:257)
  at io.objectbox.BoxStoreBuilder.build (BoxStoreBuilder.java:498)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule.providesBoxStore (DatabaseModule.java:44)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule_ProvidesBoxStoreFactory.providesBoxStore (DatabaseModule_ProvidesBoxStoreFactory.java:40)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule_ProvidesBoxStoreFactory.get (DatabaseModule_ProvidesBoxStoreFactory.java:31)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule_ProvidesBoxStoreFactory.get (DatabaseModule_ProvidesBoxStoreFactory.java:10)
  at dagger.internal.DoubleCheck.get (DoubleCheck.java:47)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule_ProvidesFetchDownloadDaoFactory.get (DatabaseModule_ProvidesFetchDownloadDaoFactory.java:35)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule_ProvidesFetchDownloadDaoFactory.get (DatabaseModule_ProvidesFetchDownloadDaoFactory.java:11)
  at dagger.internal.DoubleCheck.get (DoubleCheck.java:47)
  at org.kiwix.kiwixmobile.core.downloader.fetch.FetchDownloadMonitor_Factory.get (FetchDownloadMonitor_Factory.java:30)
  at org.kiwix.kiwixmobile.core.downloader.fetch.FetchDownloadMonitor_Factory.get (FetchDownloadMonitor_Factory.java:9)
  at org.kiwix.kiwixmobile.core.di.modules.ApplicationModule_ProvideDownloadMonitor$core_releaseFactory.get (ApplicationModule_ProvideDownloadMonitor.java:31)
  at org.kiwix.kiwixmobile.core.di.modules.ApplicationModule_ProvideDownloadMonitor$core_releaseFactory.get (ApplicationModule_ProvideDownloadMonitor.java:10)
  at dagger.internal.DoubleCheck.get (DoubleCheck.java:47)
  at org.kiwix.kiwixmobile.core.di.components.DaggerCoreComponent.injectCoreApp (DaggerCoreComponent.java:437)
  at org.kiwix.kiwixmobile.core.di.components.DaggerCoreComponent.inject (DaggerCoreComponent.java:409)
  at org.kiwix.kiwixmobile.core.CoreApp.onCreate (CoreApp.kt:85)
  at android.app.Instrumentation.callApplicationOnCreate (Instrumentation.java:1192)
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:7573)
  at android.app.ActivityThread.access$1500 (ActivityThread.java:301)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2166)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:246)
  at android.app.ActivityThread.main (ActivityThread.java:8633)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:602)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1130)
Caused by: java.lang.UnsatisfiedLinkError: 
  at java.lang.Runtime.loadLibrary0 (Runtime.java:1087)
  at java.lang.Runtime.loadLibrary0 (Runtime.java:1008)
  at java.lang.System.loadLibrary (System.java:1664)
  at io.objectbox.internal.NativeLibraryLoader.<clinit> (NativeLibraryLoader.java:84)
  at io.objectbox.BoxStore.<init> (BoxStore.java:257)
  at io.objectbox.BoxStoreBuilder.build (BoxStoreBuilder.java:498)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule.providesBoxStore (DatabaseModule.java:44)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule_ProvidesBoxStoreFactory.providesBoxStore (DatabaseModule_ProvidesBoxStoreFactory.java:40)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule_ProvidesBoxStoreFactory.get (DatabaseModule_ProvidesBoxStoreFactory.java:31)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule_ProvidesBoxStoreFactory.get (DatabaseModule_ProvidesBoxStoreFactory.java:10)
  at dagger.internal.DoubleCheck.get (DoubleCheck.java:47)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule_ProvidesFetchDownloadDaoFactory.get (DatabaseModule_ProvidesFetchDownloadDaoFactory.java:35)
  at org.kiwix.kiwixmobile.core.di.modules.DatabaseModule_ProvidesFetchDownloadDaoFactory.get (DatabaseModule_ProvidesFetchDownloadDaoFactory.java:11)
  at dagger.internal.DoubleCheck.get (DoubleCheck.java:47)
  at org.kiwix.kiwixmobile.core.downloader.fetch.FetchDownloadMonitor_Factory.get (FetchDownloadMonitor_Factory.java:30)
  at org.kiwix.kiwixmobile.core.downloader.fetch.FetchDownloadMonitor_Factory.get (FetchDownloadMonitor_Factory.java:9)
  at org.kiwix.kiwixmobile.core.di.modules.ApplicationModule_ProvideDownloadMonitor$core_releaseFactory.get (ApplicationModule_ProvideDownloadMonitor.java:31)
  at org.kiwix.kiwixmobile.core.di.modules.ApplicationModule_ProvideDownloadMonitor$core_releaseFactory.get (ApplicationModule_ProvideDownloadMonitor.java:10)
  at dagger.internal.DoubleCheck.get (DoubleCheck.java:47)
  at org.kiwix.kiwixmobile.core.di.components.DaggerCoreComponent.injectCoreApp (DaggerCoreComponent.java:437)
  at org.kiwix.kiwixmobile.core.di.components.DaggerCoreComponent.inject (DaggerCoreComponent.java:409)
  at org.kiwix.kiwixmobile.core.CoreApp.onCreate (CoreApp.kt:85)
  at android.app.Instrumentation.callApplicationOnCreate (Instrumentation.java:1192)
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:7573)
  at android.app.ActivityThread.access$1500 (ActivityThread.java:301)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2166)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:246)
  at android.app.ActivityThread.main (ActivityThread.java:8633)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:602)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1130)
```

I've not included screenshots or stack traces for crash clusters 3 to 6 to save space. They can be added if they're useful.

--- a/buildSrc/src/main/kotlin/Libs.kt
+++ b/buildSrc/src/main/kotlin/Libs.kt
@@ -420,4 +420,10 @@ object Libs {
    * https://developer.android.com/testing
    */
   const val junit: String = "androidx.test.ext:junit:" + Versions.junit
+
+  /**
+   * https://developer.android.com/reference/com/google/android/play/core/release-notes
+   */
+  const val google_android_play_core: String =
+    "com.google.android.play:core:" + Versions.google_android_play_core
 }

--- a/buildSrc/src/main/kotlin/Versions.kt
+++ b/buildSrc/src/main/kotlin/Versions.kt
@@ -12,6 +12,8 @@ import org.gradle.plugin.use.PluginDependencySpec
  */
 object Versions {
 
+  const val google_android_play_core: String = "1.7.3"
+
   const val document_file_version: String = "1.0.1"
 
   const val org_jetbrains_kotlinx_kotlinx_coroutines: String = "1.4.1"

--- a/buildSrc/src/main/kotlin/plugin/AllProjectConfigurer.kt
+++ b/buildSrc/src/main/kotlin/plugin/AllProjectConfigurer.kt
@@ -204,6 +204,7 @@ class AllProjectConfigurer {
       implementation(Libs.rxandroid)
       implementation(Libs.rxjava)
       implementation(Libs.preference_ktx)
+      implementation(Libs.google_android_play_core)
     }
   }
 }

--- a/core/src/main/java/org/kiwix/kiwixmobile/core/CoreApp.kt
+++ b/core/src/main/java/org/kiwix/kiwixmobile/core/CoreApp.kt
@@ -80,8 +80,8 @@ abstract class CoreApp : Application() {
   override fun onCreate() {
     if (MissingSplitsManagerFactory.create(this).disableAppIfMissingRequiredSplits()) {
       AlertDialog.Builder(this)
-        .setMessage(org.kiwix.kiwixmobile.core.R.string.missing_split_version_for_objectbox)
-        .setPositiveButton(R.string.yes) { dialog, which ->
+        .setMessage(R.string.missing_split_version_for_objectbox)
+        .setPositiveButton(R.string.yes) { _, _ ->
           android.os.Process.killProcess(android.os.Process.myPid())
         }
         .show()

--- a/core/src/main/java/org/kiwix/kiwixmobile/core/CoreApp.kt
+++ b/core/src/main/java/org/kiwix/kiwixmobile/core/CoreApp.kt
@@ -17,6 +17,7 @@
  */
 package org.kiwix.kiwixmobile.core
 
+import android.app.AlertDialog
 import android.app.Application
 import android.content.Context
 import android.os.Build
@@ -25,6 +26,7 @@ import android.os.Environment.getExternalStorageState
 import android.os.StrictMode
 import android.os.StrictMode.VmPolicy
 import androidx.multidex.MultiDex
+import com.google.android.play.core.missingsplits.MissingSplitsManagerFactory
 import com.jakewharton.threetenabp.AndroidThreeTen
 import org.kiwix.kiwixmobile.core.data.local.KiwixDatabase
 import org.kiwix.kiwixmobile.core.di.components.CoreComponent
@@ -76,6 +78,15 @@ abstract class CoreApp : Application() {
   }
 
   override fun onCreate() {
+    if (MissingSplitsManagerFactory.create(this).disableAppIfMissingRequiredSplits()) {
+      AlertDialog.Builder(this)
+        .setMessage(org.kiwix.kiwixmobile.core.R.string.missing_split_version_for_objectbox)
+        .setPositiveButton(R.string.yes) { dialog, which ->
+          android.os.Process.killProcess(android.os.Process.myPid())
+        }
+        .show()
+      return
+    }
     super.onCreate()
     instance = this
     coreComponent = DaggerCoreComponent.builder()

--- a/core/src/main/res/values/strings.xml
+++ b/core/src/main/res/values/strings.xml
@@ -321,4 +321,5 @@
   <string name="open_article">Open Article</string>
   <string name="delete_note_dialog_message">Note: Notes are not deleted from your storage</string>
   <string name="delete_selected_notes">Delete Selected Notes?</string>
+  <string name="missing_split_version_for_objectbox">Un-Supported architecture unable to open please install this app from PlayStore</string>
 </resources>

--- a/core/src/main/res/values/strings.xml
+++ b/core/src/main/res/values/strings.xml
@@ -321,5 +321,5 @@
   <string name="open_article">Open Article</string>
   <string name="delete_note_dialog_message">Note: Notes are not deleted from your storage</string>
   <string name="delete_selected_notes">Delete Selected Notes?</string>
-  <string name="missing_split_version_for_objectbox">Un-Supported architecture unable to open please install this app from PlayStore</string>
+  <string name="missing_split_version_for_objectbox">Un-Supported architecture, your device isn\'t compatible with this app</string>
 </resources>

